scripts/bump-version.js

```diff
@@ -6,45 +6,124 @@
+        // Assume that we're bumping the version of package that has its own independent version
+
+        const script1 = path.join(__dirname, 'independent-pkg-version.js')
+        sh.exec(`node ${script1} ${targetVersion} ${opts.package}`)
+
+        const script2 = path.join(__dirname, 'pwa-kit-deps-version.js')
+        const updateDepsBehaviour = /-dev\b/.test(targetVersion) ? 'sync' : 'latest'
+        sh.exec(`node ${script2} ${updateDepsBehaviour} ${opts.package}`)
+
+        // After updating the dependencies, let's update the package lock files
+        sh.exec('npm install')
+
+        process.exit(0)
+    }
+
+    sh.exec(`lerna version --exact --no-push --no-git-tag-version --yes ${targetVersion}`)
+    // `--exact` above is for pinning the version of the pwa-kit dependencies
+    // https://github.com/lerna/lerna/tree/main/libs/commands/version#--exact
 
-const main = () => {
-    sh.exec(`lerna version --no-push --no-git-tag-version --yes ${process.argv.slice(2).join(' ')}`)
-    sh.exec(`npm install`)
     const lernaConfig = JSON.parse(sh.cat(lernaConfigPath))
-    const rootPkg = JSON.parse(sh.cat(rootPkgPath))
-    const rootLockPkg = JSON.parse(sh.cat(rootPkgLockPath))
-
-    // find all monorepo packages, look inside each package json, find peerDependency that is a monorepo package
-    // and update it with a new version
-    const {stdout} = sh.exec('lerna list --all --json', {silent: true})
-    const packages = JSON.parse(stdout.toString())
-    const lernaPackageNames = packages.map((pkg) => pkg.name)
-    packages.forEach(({location}) => {
-        const pkgFilePath = path.join(location, 'package.json')
-        const pkg = JSON.parse(sh.cat(pkgFilePath))
-        const peerDependencies = pkg.peerDependencies
-        if (!peerDependencies) return
-        Object.keys(peerDependencies).forEach((dep) => {
-            if (lernaPackageNames.includes(dep)) {
-                console.log(`Found lerna local package ${dep} as a peer dependency of ${pkg.name}.`)
-                peerDependencies[dep] = `^${lernaConfig.version}`
-                new sh.ShellString(JSON.stringify(pkg, null, 2)).to(pkgFilePath)
-            }
-        })
-    })
+    const newMonorepoVersion = lernaConfig.version
 
     // update versions for root package and root package lock
-    rootPkg.version = lernaConfig.version
-    rootLockPkg.version = lernaConfig.version
-    new sh.ShellString(JSON.stringify(rootPkg, null, 2)).to(rootPkgPath)
-    new sh.ShellString(JSON.stringify(rootLockPkg, null, 2)).to(rootPkgLockPath)
+    setPackageVersion(newMonorepoVersion, {cwd: rootPath})
+
+    independentPackages.forEach((pkg) => {
+        const {location, version: oldVersion} = pkg
+        // Restore and then increment to the next pre-release version
+        // TODO: is it possible to _not_ trigger the lifecycle scripts? See CHANGELOG.md
+        setPackageVersion(oldVersion, {cwd: location})
+        setPackageVersion('prerelease', {cwd: location})
```

Comment:
There's a bug currently with the changelog file in commerce-sdk-react. Because the package has its own version, it will run into this code block, which will cause more headings to be added in the changelog file.

I'm thinking let's fix this bug in another PR, since it's not a blocker and the changelog can be manually fixed by hand.

![GitHub Desktop 2023-05-26 at 15 21 05](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/847300/ba8db1f5-505f-4c7e-8d79-52586029a10e)